core/src/main/java/org/mule/runtime/core/api/routing/SelectiveRouter.java

```diff
@@ -18,11 +18,11 @@
  */
 public interface SelectiveRouter extends Router {
 
-  void addRoute(String expression, MessageProcessorChain processor);
+  void addRoute(String expression, Processor processor);
```

Comment:
Discussed offline. Basically the processing strategy is created within a factory bean, so it's not possible to get a reference to it from other factory beans. That's why the chain needs to be created during initialise